### ISSUE TITLE
Memory leaks

### ISSUE THREAD

**Issue by joltcoke** [CometVisu/CometVisu#640]
I'm using an iPad 2 to show the visu. When I started using CometVisu the browser was crashing once a week due to out of memory situations. It got a lot better when jQuery and other components in CV were uploaded to the lastest version. The iPad was surviving weeks without any OOM crash.

Since the migration to the qx framework, it takes minutes, sometimes only seconds for the visu to crash. Upgraded to latest git version but same behaviour.

My start page only consist of some buttons and a weather map (image), showing only this page without switching to different pages will let the visu run for weeks. But as soon as you switch to a page with a diagram, the visu will crash fairly soon. Usually within seconds.

If I open my temperature diagram in Google Chrome on the desktop, I can see a steady increase of memory consumption as well.

**Comment by peuter** [comment]
I could reproduce this issue. A flot upgrade fixes the leak. So I don't think that the diagram memory leak is related ti the qx migration. But there might be other leaks. So please open a new issue, if there are still memory problems (after #644 got merged)

**Comment by joltcoke** [comment]
While I don't have long time experience yet, I can already say that the visu crashed twice since I upgraded last night. Maybe it's better now, but far worse than before the qa switch.

**Comment by joltcoke** [comment]
Just checked, Google Chrom is also still leaking memory.

**Comment by peuter** [comment]
Did you use a nightly build? Which one?

**Comment by peuter** [comment]
I just had the CometVisu opened with a page with 10 diagrams shown. After 2 Hours I cannot see a significant memory consumption increase (Tested with latest Chrome stable (v 61).

**Comment by joltcoke** [comment]
I'm using a self "compiled" git version. Pulled after the flot upgrade was merged. I'm using the OpenHAB mysql backend, I'm recording every value change, so there is a lot of data points shown in my diagrams.

Can I create a recoding including the diagram data transfer?

**Comment by peuter** [comment]
The diagram data should be recorded too. So give it a try. But during recording you will definitely get a memory leak, cause the record is collected "in memory". So you should download the log before the browser crashes ;-)

**Comment by peuter** [comment]
If you run your CometVisu in the same version as you recorded the log, I suggest to change that. It might not help with the memory leak, but should be a lot faster.
You are using the source version, which is mainly for development purposes. For daily use, create a build version with `./generate.py build` and replace `source` with `build` in the URL you call in your browser.

**Comment by peuter** [comment]
@joltcoke Please re-test and close this issue if the problem is solved.

**Comment by peuter** [comment]
@joltcoke Did you already had the chance to test if the problem is solved. I close this issue now. Please re-open if there is still a memory leak.
